sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala

```diff
@@ -316,7 +316,8 @@ object QueryParsingErrors {
   }
 
   def duplicateKeysError(key: String, ctx: ParserRuleContext): Throwable = {
-    new ParseException(s"Found duplicate keys '$key'.", ctx)
+    // Found duplicate keys '$key'
+    new ParseException(errorClass = "DUPLICATE_KEY_ERROR", messageParameters = Seq(key), ctx)
```

Comment:
Hard coding is used here. Looks hard to maintain. This is why I use objects:
https://github.com/apache/spark/blob/89c6f523c832daf7cff5c1ba8aecfbcd5494d5af/core/src/main/scala/org/apache/spark/errors/coreErrors.scala

Reply:
From a Scala-focused developer perspective, I agree that objects are easier to maintain. However, using string classes has a couple of benefits:
- The JSON file can be easily translated into a auto-generated documentation page (without tricks like reflection)
- Other clients (eg. Python and R) can natively throw errors with error classes, creating a single easily-auditable source of truth

What do you think? For a happy medium, we could auto-generate Scala from JSON, but I think that may increase maintenance burden.